pkg/cmd/codespace/delete.go

```diff
@@ -84,7 +84,15 @@ func (a *App) Delete(ctx context.Context, opts deleteOptions) (err error) {
 	nameFilter := opts.codespaceName
 	if nameFilter == "" {
 		a.StartProgressIndicatorWithLabel("Fetching codespaces")
-		codespaces, err = a.apiClient.ListCodespaces(ctx, api.ListCodespacesOptions{OrgName: opts.orgName, UserName: opts.userName})
+		userName := opts.userName
+		if userName == "" {
+			currentUser, err := a.apiClient.GetUser(ctx)
```

Comment:
Okay I have some more code comments; sorry for not spotting this in my last review.

I just went back to familiarize myself with ListCodespaces implementation and it turns out that ListCodespaces by default only looks up the codespaces for the currently authenticated user. In that case, what this PR introduces is an extra API call that adds overhead to populate a `UserName` field that might even not get used by ListCodespaces. The `UserName` field is only used when `OrgName` is also provided.

To avoid the unnecessary overhead in non-org mode, we could change this condition to `if userName == "" && opts.orgName != ""`, but at this point this code knows too much about ListCodespaces implementation detail.

How about we push all this logic down into ListCodespaces itself? I.e. if org name is given but username is blank, auto-fill the username of the currently authenticated user. Does that make sense?

Reply:
Yes, unfortunately moving that logic to ListCodespaces would break commands like `gh cs list --org MYORG`, so I implemented your suggestion of only looking up the current user if an org is given and the username is blank https://github.com/cli/cli/pull/6944/commits/633f03f6a6223e492101bb1a2fccd7866f89420d